mission_control/test/test_altitude_heading_behavior.py

```diff
@@ -0,0 +1,123 @@
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+"""
+import sys
+import os
+import unittest
+import math
+import rosnode
+import rospy
+import rostest
+from mission_control.msg import ReportExecuteMissionState
+from mission_control.msg import AltitudeHeading
+from auv_interfaces.msg import StateStamped
+from mission_interface import MissionInterface
+from mission_interface import wait_for
+
+
+class TestAltitudeHeadingBehavior(unittest.TestCase):
+    """ 
+        The test loads and execute an altitude heading behavior.
+        -   Load the mission altitude_heading_mission_test.xml
+        -   Execute the mission
+        -   Check if the behavior publishes the goal
+        -   Simulate auv_inteface data to finish the behavior
+        -   Test if the mission is COMPLETE
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        rospy.init_node('test_altitude_heading_behavior')
+
+    def setUp(self):
+        self.altitude_heading_goal = AltitudeHeading()
+        self.mission = MissionInterface()
+
+        self.altitude_heading_msg = rospy.Subscriber(
+            '/mngr/altitude_heading',
+            AltitudeHeading,
+            self.altitude_heading_goal_callback)
+
+        self.simulated_auv_interface_data_pub = rospy.Publisher(
+            '/state',
+            StateStamped,
+            queue_size=1)
+
+    def altitude_heading_goal_callback(self, msg):
+        self.altitude_heading_goal = msg
+        rospy.loginfo(msg)
+
+    def test_mission_with_altitude_heading_behavior(self):
+        self.mission.load_mission('altitude_heading_mission_test.xml')
+        self.mission.execute_mission()
+        self.mission.read_behavior_parameters('AltitudeHeadingBehavior')
+        altitude = self.mission.get_behavior_parameter('altitude')
+        heading = self.mission.get_behavior_parameter('heading')
+        speed_knots = self.mission.get_behavior_parameter('speed_knots')
+        enable_mask = 0
+
+        # Calculate the mask
+        def get_mask(value, mask):
+            return mask if value else 0
+        enable_mask |= get_mask(altitude, AltitudeHeading.ALTITUDE_ENA)
+        enable_mask |= get_mask(heading, AltitudeHeading.HEADING_ENA)
+        enable_mask |= get_mask(speed_knots, AltitudeHeading.SPEED_KNOTS_ENA)
+
+        def check_parameter(value):
+            return value if value else 0
+
+        def altitude_heading_goals_are_set():
+            return (self.altitude_heading_goal.altitude == check_parameter(altitude) and
+                    self.altitude_heading_goal.heading == check_parameter(heading) and
+                    self.altitude_heading_goal.speed_knots == check_parameter(speed_knots) and
+                    self.altitude_heading_goal.ena_mask == enable_mask)
```

Comment:
@patapufate same issue with type coercion, consider:
```suggestion
        def altitude_heading_goals_are_set():
            return ((altitude is None or self.altitude_heading_goal.altitude == altitude) and
                       (heading is None or self.altitude_heading_goal.heading == heading) and
                       (speed_knots is None or self.altitude_heading_goal.speed_knots == speed_knots) and
                       self.altitude_heading_goal.ena_mask == enable_mask)
```